src/component/RoadmapCanvas.js

```diff
@@ -5,117 +5,107 @@ import {AbstractRoadmapGroup} from './group/AbstractRoadmapGroup';
 
 export class RoadmapCanvas extends EventEmitter {
   type;
-  targetElement;
+  element;
   style;
 
   /**
+   * @param {Roadmap} roadmap
    * @param {RoadmapOption} option
    */
-  constructor(option) {
+  constructor(roadmap, option) {
     super();
 
+    this.roadmap = roadmap;
     this.type = option.type;
-    this.targetElement = d3.select(option.targetElementId);
+    this.element = d3.select(option.targetElementId);
     this.style = option.style;
+    this._init();
   }
 
   /**
-   * @param {Roadmap} roadmap
+   * @returns {selection}
+   * @private
    */
-  render(roadmap) {
-    this._render(roadmap, roadmap.getSortedGroup(this.type));
+  _init() {
+    this.svg = this.element.append('svg').attr('style', 'overflow: visible');
+    this.xScale = d3.scaleTime().domain([]);
+    this.yScale = d3.scaleBand().domain([]);
+    this.xAxis = this._createXAxis();
+    this.yAxis = this._createYAxis();
+    this.barArea = this._createBarArea();
   }
 
   /**
    * @param {Roadmap} roadmap
-   * @param {AbstractRoadmapGroup[]} groups
-   * @private
    */
-  _render(roadmap, groups) {
-    const tasks = groups.reduce((tasks, group) => tasks.concat(roadmap.getSortedTasksByGroup(group)), []);
+  render() {
+    const groups = this.roadmap.getSortedGroups(this.type);
+    const tasks = this.roadmap.getGroupSortedTasks(this.type);
 
-    const svg = this._appendSVG();
-    const marginBottom = this._getXAxisHeight(svg, tasks);
-    const marginLeft = this._getYAxisWidth(svg, groups);
+    const marginBottom = this._getXAxisHeight(tasks);
+    const marginLeft = this._getYAxisWidth(groups);
 
-    const w = this.targetElement.node().clientWidth - marginLeft;
+    const w = this.element.node().clientWidth - marginLeft;
     const h = (tasks.length * this.style.barHeight) + (tasks.length + 1) * this.style.barHeight * this.style.barPadding;
-    svg
+
+    // styling.
+    this.svg
       .attr('width', w)
       .attr('height', h)
       .attr('style', `padding: 0 0 ${marginBottom}px ${marginLeft}px`);
 
     // create chart.
-    const xScale = this._generateXAxisScale(tasks, w);
-    const yScale = this._generateYAxisScale(tasks, h);
-    this._appendXAxis(svg, xScale, h);
-    this._appendYAxis(roadmap, groups, svg, yScale);
-    this._appendTaskLines(tasks, svg, xScale, yScale);
+    this._updateXScale(tasks, w);
+    this._updateYScale(tasks, h);
+    this._updateXAxis(w, h);
+    this._updateYAxis();
+    this._updateTaskBars(tasks);
 
     // utility.
-    this._addMouseHelper(roadmap, svg, xScale, yScale, marginLeft);
+    // this._addMouseHelper(marginLeft);
```

Comment:
Sorry...I was disable this feature.
When drag&drop, mouse helper dupulicate draw lines...